File: vibe-lang/benchmark/output/python-pygame/gemini/health_bar.py
import pygame
import sys

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Health Bar System")
clock = pygame.time.Clock()
font = pygame.font.Font(None, 36)

# Game state variables
health = 100.0
max_health = 100.0
regen_rate = 5.0

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False
        if event.type == pygame.KEYDOWN:
            if event.key == pygame.K_SPACE:
                health -= 15.0
                if health < 0.0:
                    health = 0.0

    # Update logic
    if health > 0.0 and health < max_health:
        health += regen_rate * dt
        if health > max_health:
            health = max_health

    # Draw logic
    screen.fill((0, 0, 0))
    
    # Draw health bar background
    pygame.draw.rect(screen, (100, 100, 100), (250, 50, 300, 30))
    
    # Draw filled portion
    bar_width = (health / max_health) * 300.0
    pygame.draw.rect(screen, (0, 255, 0), (250, 50, int(bar_width), 30))
    
    # Display HP text
    hp_text = font.render(f"HP: {int(health)} / 100", True, (255, 255, 255))
    screen.blit(hp_text, (250, 90))
    
    # Draw player
    pygame.draw.rect(screen, (255, 255, 255), (384, 300, 32, 32))
    
    # Display instructions
    instr_text = font.render("Press SPACE to take damage", True, (255, 255, 255))
    screen.blit(instr_text, (230, 550))

    pygame.display.flip()

pygame.quit()
sys.exit()

File: vibe-lang/benchmark/output/lua-love/gemini/health_bar.lua
local health = 100.0
local max_health = 100.0
local regen_rate = 5.0

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Health Bar System")
end

function love.update(dt)
    if health > 0.0 and health < max_health then
        health = health + regen_rate * dt
    end
    
    if health > max_health then
        health = max_health
    end
end

function love.keypressed(key)
    if key == "space" then
        health = health - 15.0
        if health < 0.0 then
            health = 0.0
        end
    end
end

function love.draw()
    -- Draw background bar
    love.graphics.setColor(0.3, 0.3, 0.3)
    love.graphics.rectangle("fill", 250.0, 50.0, 300.0, 30.0)
    
    -- Draw filled portion
    local bar_width = (health / max_health) * 300.0
    love.graphics.setColor(0.0, 1.0, 0.0)
    love.graphics.rectangle("fill", 250.0, 50.0, bar_width, 30.0)
    
    -- Reset color and draw text
    love.graphics.setColor(1, 1, 1)
    love.graphics.print("HP: " .. math.floor(health) .. " / 100", 250.0, 90.0)
    
    -- Draw player
    love.graphics.rectangle("fill", 384.0, 300.0, 32, 32)
    
    -- Draw instruction
    love.graphics.print("Press SPACE to take damage", 230.0, 550.0)
end